src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -2314,6 +2314,53 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/search/{id}/metadata:
+    post:
+      tags:
+        - search
+        - snapshots
+        - repository
+      description: Insert or update search metadata for a snapshot
+      operationId: upsertSearchMetadata
+      parameters:
+        - name: id
+          in: path
+          description: A UUID to used to identify an object in the repository
+          required: true
+          schema:
+            type: string
```

Comment:
Could use this `UniqueIdProperty` so the incoming object is `UUID`?

Reply:
We also have an `Id` parameter that is used by several OpenAPI endpoints, I am not sure which one should be used preferentially.

Reply:
I ended up using `UniqueIdProperty`